Update guessit to 2.0

Toilal opened the pull request:
Work in progress...

This is the first attempt for #550. There are a few tests failing, I don't have time to fix them for now, but feel free to comment the PR !

Diaoul commented on subliminal/providers/opensubtitles.py:
```diff
@@ -66,7 +66,7 @@ def get_matches(self, video, hearing_impaired=False):
                 matches.add('title')
             # guess
-            matches |= guess_matches(video, guess_episode_info(self.movie_release_name + '.mkv'))
+            matches |= guess_matches(video, guessit(self.movie_release_name + '.mkv', {'type': 'episode'}))
```
Do we still need the extension?

Toilal replied:
No, it can be dropped.
Le 26 janv. 2016 23:48, "Antoine Bertin" <notifications@github.com> a
écrit :

> In subliminal/providers/opensubtitles.py
> <https://github.com/Diaoul/subliminal/pull/564#discussion_r50914250>:
>
> > @@ -66,7 +66,7 @@ def get_matches(self, video, hearing_impaired=False):
> >              if video.title and sanitized_string_equal(self.series_title, video.title):
> >                  matches.add('title')
> >              # guess
> > -            matches |= guess_matches(video, guess_episode_info(self.movie_release_name + '.mkv'))
> > +            matches |= guess_matches(video, guessit(self.movie_release_name + '.mkv', {'type': 'episode'}))
>
> Do we still need the extension?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Diaoul/subliminal/pull/564/files#r50914250>.
>

Diaoul commented on subliminal/video.py:
```diff
@@ -96,7 +96,7 @@ def fromguess(cls, name, guess):
 
         :param str name: name of the video.
-        :param dict guess: guessed data, like a :class:`~guessit.guess.Guess` instance.
+        :param dict guess: guessed data, like a :class:`~rebulk.match.MatchesDict` instance.
```
I believe this would require to update the docs so it shows proper linking to rebulk's docs.

Diaoul replied:
Here is the portion of code to edit: https://github.com/Diaoul/subliminal/blob/master/docs/conf.py#L300-L309

Toilal replied:
Currently I don't have docs published with sphinx on rebulk. Some maybe I could juste drop this link ?

Diaoul commented on subliminal/subtitle.py:
```diff
@@ -276,13 +275,16 @@ def guess_matches(video, guess, partial=False):
             matches.add('series')
+        # series
+        if video.title and 'episode_title' in guess and sanitized_string_equal(guess['episode_title'], video.title):
+            matches.add('title')
```
I prefered the series/title over title/episode_title. From an episode standpoint (which is what a video file is), the title is the title of the episode, not the title of the series.
I believe I already said that somewhere in guessit's bugtracker but both points are valid so it doens't really matter.

Toilal replied:
Yes => https://github.com/guessit-io/guessit/issues/232#issuecomment-154168664

Doesn't really matter though, I've finally make this naming choice because it makes guessit codebase more concise and less confuse.

Diaoul replied:
Fine by me

labrys replied:
Just my two cents worth, I find the series/title a better format too and it still works for movies as the "series" title for a movie can be considered the set title, e.g. `Back to the Future Trilogy` and `Back to the Future Part III` however i can see how this could potentially complicate the code unnecessarily.

Toilal replied:
guessit 2 will find one of them as `alternative_title`. Search for alternative_title in this file, and you will have some examples: https://github.com/guessit-io/guessit/blob/master/guessit/test/movies.yml

Keep in mind than guessit is not a "human", and it's just guessing :) So it's actually impossible to find out that a particular string is an alternative title, a movies series title (like James Bond), or even a productor or main actor of the movie ... It will just go into `alternative_title` list.

Diaoul commented:
Thanks for the PR, I left some inline comments. I can see some tests are failing, I'll look into it tomorrow.

Toilal commented:
Ok, i've fixed remaining tests and made some changes with comments you have made (removed .mkv extensions, removed invalid link to inexistant rebulk docs). It should works now !

Toilal commented:
Do you plan a release after merging this one ? Would be great for Flexget and SickRage projects :p

Toilal commented:
There are still issues with python 2.7,i'll fix them. I'll also release guessit as 2.0 final.

Toilal commented:
I'm currently working on proper string support in both python 2 & python 3 in guessit. This will fix remaining failing test.

Toilal commented:
All is OK now !

Toilal commented:
Requirements should be >=2.0.3 because of an issue with unicode handling.
